### PR TITLE
Edit Exhisting Notes

### PR DESCRIPTION
**Pull request by laurengordonfahn**
Enabling users to edit notes and update the database with the new content.

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -38,6 +38,22 @@ def add_note():
 
     return jsonify(response)
 
+@app.route('/notes/edit/<id>', methods=['POST'])
```
Comment (boz) [inline server.py]: The standard route for this would be `PUT /notes/<id>`.

File: server.py [hunk 2]
```diff
@@ -38,6 +38,22 @@ def add_note():
...
+    note_title = request.form.get('note_title')
+    note_content = request.form.get('note_content')
+
+    
+    note = update_note(note_id, note_title, note_content)
+    print "update is running RUNRUNRUNRUN", note.content, note.created_at
+
+    flash("Your Edits Have Been Saved")
+
+    return redirect('/')
```
Comment (boz) [inline server.py]: Get rid of the `print`.

I only see this being called via AJAX.  What does `flash()` do in this case?  How is the 3xx response handled?  Does it think it's an error?  Does it follow the redirect and return the contents of `/` to your callbacks?

**Review by boz — CHANGES_REQUESTED** [review]
Looking good.  See comments.

File: helper_functions.py [hunk 3]
```diff
@@ -19,6 +19,19 @@ def commit_note_to_db(note_title, new_note):
...
+def update_note(note_id, note_title, note_content):
+    """ Update a note By Id Return Updated Note Object """
+
+    # note = Note.query.filter_by(id=note_id).update(dict(title=note_title, content=note_content))
+    note = Note.query.filter_by(id=note_id).first()
+    note.title = note_title
+    note.content = note_content
+    
+   
+    db.session.commit()
```
Comment (boz) [inline helper_functions.py]: what happens if there's not a note with `id=note_id`?

do the changes actually get saved here?

File: static/javascript/index.js [hunk 4]
```diff
@@ -33,6 +33,68 @@ function addNewNoteToDB(event){
...
+
+    
+    //TODO fix path to 
+    $.ajax({
+        url: "/notes/edit/"+ note_id,
+        type: 'POST',
+        data: {
+        "note_id" : note_id,
+        "note_title": note_title,
+        "note_content": note_content
```
Comment (boz) [inline static/javascript/index.js]: Send the note to the server using the same format that you receive the note from the server (the keys shouldn't have "note_" prefixes).

File: server.py (hunk 1 above)
Comment (laurengordonfahn) [reply server.py]: When I use PUT I get an error messages: 405 Method not allowed. I have googled and don't see a resolution to this issue. Requires further research.

File: server.py (hunk 2 above)
Comment (laurengordonfahn) [reply server.py]: At the moment is does post a message and follows the redirect to / but not properly as it is repeated with 3xx response. I will look into this.

File: server.py (hunk 1 above)
Comment (boz) [reply server.py]: were you allowing the `PUT` method in the route?

**Comment by laurengordonfahn** [comment]
Yes in both the route and the callback.


Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Adam Bozanich <notifications@github.com> Date: 2/3/17  6:36 PM  (GMT-08:00) To: laurengordonfahn/simple_notebook_project <simple_notebook_project@noreply.github.com> Cc: Lauren Gordon-Fahn <laurengordonfahn@gmail.com>, Author <author@noreply.github.com> Subject: Re: [laurengordonfahn/simple_notebook_project] Edit Exhisting Notes (#18) 
@boz commented on this pull request.



In server.py:
> @@ -38,6 +38,22 @@ def add_note():
 
     return jsonify(response)
 
+@app.route('/notes/edit/<id>', methods=['POST'])

were you allowing the PUT method in the route?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/laurengordonfahn/simple_notebook_project","title":"laurengordonfahn/simple_notebook_project","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/laurengordonfahn/simple_notebook_project"}},"updates":{"snippets":[{"icon":"PERSON","message":"@boz commented on #18"}],"action":{"name":"View Pull Request","url":"https://github.com/laurengordonfahn/simple_notebook_project/pull/18"}}}

**Pull request merged**